infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -342,7 +342,7 @@ private synchronized void runDetectionTasks() {
             Layout updatedLayout = correctWrongEpochs(pollReport, layout);
 
             Result<DetectorTask, RuntimeException> failure = Result.of(() -> {
-                if (pollReport.isCurrentLayoutSlotUnFilled()) {
+                if (pollReport.getLayoutSlotUnFilled(updatedLayout).isPresent()) {
                     detectFailure(updatedLayout, Collections.emptySet(), pollReport).join();
```

Comment:
Can we add a log statement here to know that this layout slot was filled with the previous layout and no new cluster reconfigurations have taken place?

Reply:
Ack.